client/catapult/tests/catapult/model/TransactionTests.cpp

```diff
@@ -118,8 +118,18 @@ namespace catapult { namespace model {
 		// Arrange:
 		std::vector<uint8_t> buffer(sizeof(SizePrefixedEntity));
 		auto* pTransaction = reinterpret_cast<Transaction*>(&buffer[0]);
+
+#if defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Warray-bounds" // Transaction[0] is partly outside array bounds
+#endif
+
 		pTransaction->Size = sizeof(SizePrefixedEntity);
```

Comment:
warning is because buffer.size() is smaller than sizeof(Transaction). alternatively, could make buffer larger, but that somewhat weakens the test.